aws-ts-static-website/index.ts

```diff
@@ -79,6 +78,38 @@ const logsBucket = new aws.s3.Bucket("requestLogs",
 
 const tenMinutes = 60 * 10;
 
+let certificateArn: pulumi.Input<string> = config.certificateArn!;
+if (config.certificateArn === undefined) {
+    // Per AWS, ACM certificate must be in the us-east-1 region.
+    const eastRegion = new aws.Provider("east", {
+        region: "us-east-1",
+    });
+    const certificate = new aws.acm.Certificate("certificate", {
+        domainName: config.targetDomain,
+        validationMethod: "DNS",
+    }, { provider: eastRegion });
+
+    const domainParts = getDomainAndSubdomain(config.targetDomain);
+    const hostedZoneId = aws.route53.getZone({ name: domainParts.parentDomain }).then(zone => zone.id);
+
+    const certificateValidationDomain = new aws.route53.Record(
```

Comment:
A comment to better explain what is going on here is probably warranted. What you are doing is creating the ACM certificate, that resource then has the `domainValidationOptions`, which you then use to create a new DNS record, which is used to verify that you own the domain. Right?

It's cool that it all "just works", but some docs explaining what's going on would be useful for the next person who looks at this code scratching their head.

Reply:
Yep, great point.